### PR TITLE
feat(worker): deterministic worker working directory

### LINKED ISSUE #3584: feat(worker): deterministic worker working directory
Actually some tools requires to be ran with the directory structures.

**Example from terraform**:
https://www.terraform.io/guides/running-terraform-in-automation.html 

Terraform currently makes some assumptions which must be accommodated by such an automation setup:

The saved plan file can contain absolute paths to child modules and other data files referred to by configuration. Therefore it is necessary to ensure that the archived configuration is extracted at an identical absolute path. This is most commonly achieved by running Terraform in some sort of isolation, such as a Docker container, where the filesystem layout can be controlled.

but out worker working directory is not deterministic:
```go
// Setup working directory
	pbJobPath := path.Join(fmt.Sprintf("%d", jobInfo.Number),
		fmt.Sprintf("%d", jobInfo.SubNumber),
		fmt.Sprintf("%d", jobInfo.NodeJobRun.ID),
		fmt.Sprintf("%d", jobInfo.NodeJobRun.Job.PipelineActionID))
	wd := workingDirectory(w.basedir, pbJobPath)
```

we should have to refactor this to be able to have always the same working directory (for the same workermodel)

### PR DESCRIPTION
**Pull request by fsamin**
1. Description
Set the working directory from base64 encoded job name
1. Related issues
closes #3584
1. About tests

1. Mentions

@ovh/cds

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#7826.1 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✘  
  * tools/debpacker ✘  
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report build-ui#7826.2 ✘  
* Compile
  * Build UI ✘  
  * TU All -   
  * TU model ✔ 
  * TU service ✔ 
  * TU shared ✘  
  * TU views ✔

**Comment by ovh-cds** [comment]
CDS Report ut-engine#7826.2 ✘  
* Stage 1
  * Engine Test with DB ✘

File: engine/worker/run.go [hunk 1]
```diff
@@ -341,21 +340,16 @@ func teardownBuildDirectory(wd string) error {
...
-	if _, err := rand.Read(bs); err != nil {
-		return "", err
-	}
-	str := hex.EncodeToString(bs)
-	token := []byte(str)[0:size]
-
-	return string(token), nil
-}
+func workingDirectory(basedir string, jobInfo *sdk.WorkflowNodeJobRunData, suffixes ...string) string {
+	var encodedName = base64.StdEncoding.EncodeToString([]byte(jobInfo.NodeJobRun.Job.Job.Action.Name))
```
Comment (richardlt) [inline engine/worker/run.go]: With the local hatchery case, two or more worker will maybe share the same workspace.

Comment (richardlt) [inline engine/worker/run.go]: With the local hatchery case, two or more worker will maybe share the same workspace.

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#7847.0 ✘  
* Build minimal
  * cdsctl linux ✘  
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✔ 
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Comment by ovh-cds** [comment]
CDS Report build-engine-cli-tools#7848.0 ✘  
* Build minimal
  * cdsctl linux ✔ 
  * Engine Build linux amd64 ✔ 
  * tools/debpacker ✘  
  * tools/os-ansible-inventory ✔ 
  * Worker linux amd64 / 386 ✔

**Pull request merged**